### DISCUSSION TITLE
Add unification to `Error`

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#8477] [Ideas]
For this program:
```
def example(x: , y: Int32): Int32 = x + y
```
instead of `x` having type `Error`, it would be much more useful to have it be type `Error(but really Int32)`.

This would mean that
- you can see types of wrong expressions (e.g. `asd` is undefined, but should have type `Int32`)
- you can use this in vscode to fill in the type of `x` in the above program, since the compiler gives `Error(Int32)` back

**Comment by magnus-madsen** [comment]
How did you know it should have type Int32? Is this just a proposal for type inference at the top-level in disguise? :)

**Comment by JonathanStarup** [comment]
`x` is used in `x + y` where `y: Int32` and `Add.add` requires two equal types

It is something yes, but this helps a lot of other cases too. e.g. this program above has the error that `y` has the wrong type! it should have type `Error` so you can `Add.add` it to `x`.

Treating Error as a type variables with a flag seems less fragile (see how many times it crashes if you put it in formulas or associated effects)
